sql/catalyst/src/test/scala/org/apache/spark/sql/RowTest.scala

```diff
@@ -121,10 +121,6 @@ class RowTest extends FunSpec with Matchers {
       externalRow should be theSameInstanceAs externalRow.copy()
     }
 
-    it("copy should return same ref for internal rows") {
```

Comment:
As long as `InternalRow` can have mutable implementation, `InternalRow` is not immutable anymore, because it can have a struct field, whose value can be a `MutableRow`.